addon/globalPlugins/webAccess/gui/properties.py

```diff
@@ -0,0 +1,685 @@
+		("sayName", pgettext("webAccess.ruleProperties", "&Speak rule name")),
+		# Translator: Custom name input label for the rule dialog's properties panel.
+		("customName", pgettext("webAccess.ruleProperties", "Custom &name:")),
+		# Label depends on rule type)
+		("customValue", None),
+		# Translator: Transform select label for the rule dialog's properties panel.
+		("mutation", pgettext("webAccess.ruleProperties", "&Transform:")),
+	))
+
+RULE_TYPE_FIELDS = OrderedDict((
+	(ruleTypes.PAGE_TITLE_1, ("customValue",)),
+	(ruleTypes.PAGE_TITLE_2, ("customValue",)),
+	(ruleTypes.ZONE, (
+		"autoAction"
+		"formMode",
+		"skip",
+		"sayName",
+		"customName",
+		"customValue",
+		"mutation"
+	)),
+	(ruleTypes.MARKER, (
+		"autoAction"
+		"multiple",
+		"formMode",
+		"skip",
+		"sayName",
+		"customName",
+		"customValue",
+		"mutation"
+	)),
+	))
+
+def showPropsDialog(context, properties):
+	PropsMenu(context, properties).ShowModal()
+	return True
+
+class ListControl(object):
+
+	def __init__(
+			self,
+	        *args
+	):
+		# wx.List control elements from client side
+		super(ListControl, self).__init__()
+		self.propsPanel = args[0]
+		self.propertiesList = args[1]
+		self.listCtrl = args[2]
+		self.toggleBtn = args[3]
+		self.editable = args[4]
+		self.choice = args[5]
+		self.btnAddProps = args[6]
+		self.btnDelProps = args[7]
+		self.context = args[8]
+
+		# wx.List control event binding
+		self.listCtrl.Bind(wx.EVT_LIST_ITEM_SELECTED, self.onItemSelListCtrl)
+		self.listCtrl.Bind(wx.EVT_KEY_DOWN, self.onKeyPress)
+		self.toggleBtn.Bind(wx.EVT_TOGGLEBUTTON, self.updateValueEvtToggle)
+		self.editable.Bind(wx.EVT_TEXT, self.updateValueEvtEdit)
+		self.choice.Bind(wx.EVT_CHOICE, self.updateValueEvtChoice)
+		self.btnAddProps.Bind(wx.EVT_BUTTON, self.onAddProperties)
+		self.btnDelProps.Bind(wx.EVT_BUTTON, self.onDeleteProperties)
+
+		#Local variables
+		self.index = 0
+		self.incr = 0
+		self.currentSelItem = None
+		self.mutationOptions = []
+		self.autoActionOptions = []
+		self.formmodeOptions = ("", "unchanged", "Enable", "disable")
+
+		#Instanciation of increment values for wx.choice and updates the correspondant wx.listCtrl
+		self.objIncAutoAct = IncrementValue()
+		self.objIncMut = IncrementValue()
+		self.objIncFormMde = IncrementValue()
+		self.init()
+
+	def init(self):
+		self.toggleBtn.Disable()
+		self.editable.Disable()
+		self.choice.Disable()
+		self.updateBtnState()
+		self.getAutoActions()
+
+	def messageBox(self, message, caption):
+		gui.messageBox(
+			# Translators: The text of a generic error message dialog
+			message=_("{}".format(message)),
+			# Translators: The title of an error message dialog
+			caption=_("{}".format(caption)),
+			style=wx.OK | wx.ICON_EXCLAMATION
+		)
+
+	def interpretBoolVal(self, val):
+		ret = lambda x: 'Actif' if x == True else ('Inactif' if x == False else x)
+		retVal = lambda x: ret(x) if type(x) == bool else ("vide" if x == None else x)
+		return retVal(val)
```

Comment:
Needlessly complex, imho.

```suggestion
	def interpretBoolVal(self, val):
		if val is True:
			return _("Enabled")
		elif val is False:
			return _("Disabled")
		elif val is None:
			return _("Empty")
		return val
```